docs/api/session.md

```diff
@@ -879,6 +879,9 @@ session.fromPartition('some-partition').setPermissionCheckHandler((webContents,
         Specifying a loopback device will capture system audio, and is
         currently only supported on Windows. If a WebFrameMain is specified,
         will capture audio from that frame.
+      * `disableLocalEcho` Boolean (optional) - If `audio` is a [WebFrameMain](web-frame-main.md)
+         and if this is set to `true`, then prevents input from being echoed in the local output
+         stream (e.g., speaker). Defaults to `true`.
```

Comment:
Generally we prefer to phrase boolean options in the positive unless there's a good reason to do otherwise. Might this parameter be clearer if it were named `enableLocalEcho`, with a default value of `false`?

Reply:
I'd prefer it as you've suggested, but that would conflict with `disable_local_echo = true` above

Reply:
you mean in the C++ code?

Reply:
I removed my comment as I realized the inverse can still work.

Reply:
Actually, I take that back. I've revised it to:

```
`enableLocalEcho` Boolean (optional) - If `audio` is a [WebFrameMain](web-frame-main.md)
         and if this is set to `false`, then prevents input from being echoed in the local output
         stream (e.g., speaker). Defaults to `false`.
```

Then the C++ code becomes

```
    } else if (result_dict.Get("audio", &rfh)) {
      bool disable_local_echo = true;
      result_dict.Get("enableLocalEcho", &disable_local_echo);
```

That's incorrect though unless there's a way to inverse the value in the `Get` call when `enableLocalEcho` is defined.

Reply:
@nornagon in your PR for implementing `setDisplayMediaRequestHandler`

https://github.com/electron/electron/pull/30702#discussion_r929303621

You noted that you were actually unsure of the functionality of the loopback param and it could possibly be tweaked.

It's possible your default choice of `true` is the correct one, and would be concerned that users might report bugs around audio echoing should the default be swapped to `false`.

Reply:
I believe I decided not to expose this because in Chrome it wasn't generally possible to alter that setting from Web content.

As for inverting the sense of this boolean, this should work fine:

```cpp
    } else if (result_dict.Get("audio", &rfh)) {
      bool enable_local_echo = false;
      result_dict.Get("enableLocalEcho", &enable_local_echo);
      bool disable_local_echo = !enable_local_echo;
```

Reply:
Compiler did not like that:

```
    } else if (result_dict.Get("audio", &rfh)) {
      bool enable_local_echo = false;
      result_dict.Get("enableLocalEcho", &enable_local_echo);
      bool disable_local_echo = !enable_local_echo;
      devices.audio_device = blink::MediaStreamDevice(
          request.audio_type,
          content::WebContentsMediaCaptureId(rfh->GetProcess()->GetID(),
                                             rfh->GetRoutingID(),
                                             disable_local_echo)
```                                             

```
../../electron/shell/browser/electron_browser_context.cc:507:12: error: declaration shadows a local variable [-Werror,-Wshadow]
      bool disable_local_echo = !enable_local_echo;
           ^
../../electron/shell/browser/electron_browser_context.cc:495:10: note: previous declaration is here
    bool disable_local_echo;
         ^
../../electron/shell/browser/electron_browser_context.cc:495:10: error: unused variable 'disable_local_echo' [-Werror,-Wunused-variable]
    bool disable_local_echo;
```

Reply:
remove your earlier, now unused declaration of `bool disable_local_echo`

Reply:
> remove your earlier, now unused declaration of `bool disable_local_echo`

Forgot about that var, thanks.